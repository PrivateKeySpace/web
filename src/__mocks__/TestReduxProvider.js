import React from 'react'
import PropTypes from 'prop-types'
import { createStore } from 'redux'
import { Provider as ReduxProvider } from 'react-redux'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'
import rootReducer from '../store/rootReducer'

const displayName = '__mocks__/TestReduxProvider'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ state: PropTypes.object, children: PropTypes.any }),
  defaultProps({ state: {} })
)

function TestReduxProvider (props) {
  const { state, children } = props
  const store = createStore(rootReducer, state)

  return (
    <ReduxProvider store={store}>
      {children}
    </ReduxProvider>
  )
}

export default enhance(TestReduxProvider)
