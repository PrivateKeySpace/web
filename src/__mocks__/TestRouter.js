import React from 'react'
import PropTypes from 'prop-types'
import { MemoryRouter } from 'react-router'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'

const displayName = '__mocks__/TestRouter'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ path: PropTypes.string, children: PropTypes.any }),
  defaultProps({ path: '/' })
)

function TestRouter (props) {
  const { path, children } = props

  return (
    <MemoryRouter initialEntries={[path]} initialIndex={0}>
      {children}
    </MemoryRouter>
  )
}

export default enhance(TestRouter)
