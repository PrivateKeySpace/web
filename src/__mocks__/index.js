export { default as TestRouter } from './TestRouter'
export { default as TestReduxProvider } from './TestReduxProvider'
