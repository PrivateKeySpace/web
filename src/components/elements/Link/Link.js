import { Link as RouterLink } from 'react-router-dom'
import { compose, setDisplayName, setPropTypes, mapProps } from 'recompose'
import { propTypes, mapProps as propsMapper } from './lib'

const displayName = 'components/elements/Button'
const enhance = compose(
  setDisplayName(displayName),
  setPropTypes(propTypes),
  mapProps(propsMapper)
)

export default enhance(RouterLink)
