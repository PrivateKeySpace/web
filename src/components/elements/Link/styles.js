import { StyleSheet } from 'aphrodite/no-important'
import { colors } from '../../../theme/constants'
import { darken } from 'polished'

const styles = StyleSheet.create({
  link: {
    color: colors.info,
    cursor: 'pointer',
    ':hover': {
      color: darken(0.1, colors.info)
    }
  }
})

export default styles
