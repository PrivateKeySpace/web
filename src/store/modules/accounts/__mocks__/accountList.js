const accountList = [
  {
    'path': [
      2147483697,
      2147483648,
      2147483648
    ],
    'address': '3Pc8vh9xUopMmiBW1yDAm3h1aqqbmmv11p',
    'addressPath': [
      2147483697,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6Ca3FEzN1SemuKNujpAeyDLXk7DXXtgpUs5HHLrG5brjVQm7ib3SLqMKfWWQZset7HMm7wM58wVGEJm5eG6rkxEcYvtvExYaM9T7jxkmmMp',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': true
  },
  {
    'path': [
      2147483692,
      2147483648,
      2147483648
    ],
    'address': '1Bswvjfoa1p1kMkwE4C8KAPbYLSKfg1Nni',
    'addressPath': [
      2147483692,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6D4zdJZuWGKQK5qMyj7reDcapRE6StaVTGS3xWCAjRyLfne84tGxUK1yYNrUWFY8gYtJA7xd3s97o1uiPkFEhpdsuDsNkLnj5iHyAbCEnRv',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': false
  }
]

export default accountList
