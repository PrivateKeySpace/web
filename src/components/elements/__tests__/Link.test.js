import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import { TestRouter } from '../../../__mocks__'
import Link from '../Link'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders without crashing and matches snapshot', () => {
  const link = render(
    <TestRouter path='/'>
      <Link to='/test'>Test link</Link>
    </TestRouter>
  )
  expect(link).toMatchSnapshot()
})
