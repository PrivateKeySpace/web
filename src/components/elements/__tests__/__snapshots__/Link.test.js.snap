// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<a
  class="textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-fontSize3_o9hx2m-o_O-link_1etlbkf"
  href="/test"
>
  Test link
</a>
`;
